### ISSUE TITLE
Inverted option OptionAutoCloseOnEnd

### ISSUE THREAD

**Issue by Angelina88** [zzzprojects/html-agility-pack#88]
I've noticed that this option works a bit strange. When it is false an html like "....<title>.....<title>...." passed without errors and a closing tag for <title> was not added. I set the option to true and get correct parse error without auto fixing.

Thanks,
Angelina

**Comment by JonathanMagnan** [comment]
Hello @Angelina88 ,

Thank you for reporting, I will try to look at it today or tomorrow.

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello @Angelina88 ,

Could you give me a working example? with the expected result

By example:

```csharp
var htmlDoc = new HtmlAgilityPack.HtmlDocument() { OptionAutoCloseOnEnd = false};
htmlDoc.LoadHtml("<title>def<title>abc</title><div>a</div>");

string s = htmlDoc.DocumentNode.InnerHtml;
```
I know the current v1.x have some issue for correctly ending some tag but we are currently working on it (for v2.x).

I want to be sure to fix the right thing.

Best Regards,

Jonathan

**Comment by Angelina88** [comment]
Hello, @JonathanMagnan 
Here example:
```
var htmlDoc = new HtmlAgilityPack.HtmlDocument() { OptionAutoCloseOnEnd = false };
            htmlDoc.LoadHtml(@"
<!DOCTYPE html PUBLIC "" -//W3C//DTD XHTML 1.0 Transitional//EN"" ""http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"">
<html xmlns = ""http://www.w3.org/1999/xhtml"">
    <head>
        <meta http - equiv = ""Content-Type"" content = ""text/html; charset=utf-8"" />
        <title> A surprise gift for your birthday<title>  
    </head>
    <body>

                         
    </body>
</html>
");

            string s = htmlDoc.DocumentNode.InnerHtml;
            //title should not be added, but it is there
```

**Comment by JonathanMagnan** [comment]
Thank you @Angelina88 ,

In fact, if I fix it correctly, the output will be even weirder since the `title` must be closed by an ending `title` otherwise it includes everything inside as text such as:

```text
<html xmlns="http://www.w3.org/1999/xhtml"><head>
        <meta http="" -="" equiv="Content-Type" content="text/html; charset=utf-8">
        <title> A surprise gift for your birthday&lt;title&gt;
    &lt;/head&gt;
    &lt;body&gt;

                         
    &lt;/body&gt;
&lt;/html&gt;</title></head><body></body></html>
```

To minimize side impact until v2.x come out, I have fixed the title to automatically end when a starting title is found such as this output:

```text

<!DOCTYPE html PUBLIC " -//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
    <head>
        <meta http="" -="" equiv="Content-Type" content="text/html; charset=utf-8">
        <title> A surprise gift for your birthday</title><title>  
    </title></head>
    <body>

                         
    </body>
</html>
```

The second title is still here but there is not a lot of stuff I can do here since the head tag can have multiple title tag even if invalid.

The fix should be available for next Monday,

Let me know if that could solve your problem.

Best Regards,

Jonathan

**Comment by Angelina88** [comment]
@JonathanMagnan ,
But if OptionAutoCloseOnEnd = false titles should stay without a closing tag, shouldn't it?

Best regards,
Angelina

**Comment by JonathanMagnan** [comment]
Nope I don't think it works like this.

Honestly, I'm not quite sure how this option work.

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello @Angelina88 ,

The v1.6.2 has been released

We improved the way we handle `title`, it's not quite perfect yet but we hope to have all this kind of issue fixed in the v2.x

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello @Angelina88 ,

We will close this issue since it should be solved in v1.6.2

Best Regards,

Jonathan

---
_Help us to keep this library free: [Donate](http://www.zzzprojects.com/contribute)_
